lib/Reader/readerir.cpp

```diff
@@ -1932,6 +1904,52 @@ void GenIR::createOverlapFields(
   OverlapFields.clear();
 }
 
+char *GenIR::getClassNameWithNamespace(CORINFO_CLASS_HANDLE ClassHandle) {
+  // Fetch the name of this type.
+  // Note some constructed types like arrays may not have names.
+  int32_t NameSize = 0;
+  const bool IncludeNamespace = true;
+  const bool FullInst = false;
+  const bool IncludeAssembly = false;
+  // We are using appendClassName instead of getClassName because
+  // getClassName omits namespaces from some types (e.g., nested classes).
+  // We may still get the same name for two different classes because
+  // two classes with the same fully-qualified names may live in different
+  // assemblies.
```

Comment:
What will happen if we get the same name for different classes, will that cause an error?  If so, would fixing it be as simple as passing true for IncludeAssembly here?